### DISCUSSION TITLE
Query multiple datasets in export template

### DISCUSSION THREAD

**Discussion by Decafpancakes** [netbox-community/netbox#9468] [Q&A]
I have been messing around with the export template and i have 2 templates. one to export the VLANS and one to export the IPAddresses. What I'm trying to do is combine the two so that IPs within the particular VLAN prefix are exported with the VLAN config. I thought this might be possible by getting the prefix for the VLAN then getting the child ips within it but so far I cant figure out how to reference it. Both of these exports work great independently, just not together. So far this is the export code:

 IP:
{% for IPAddress in queryset %}
Desc: , Location: 
host {{ IPAddress.dns_name}} { hardware ethernet {{ IPAddress.cf.mac_address }}; fixed-address {{ IPAddress.address }}; }

{% endfor %}

VLAN: 
{% for vlan in queryset %}
#### VLAN {{vlan.vid}} ####

#$ {{vlan.cf.domain_name}}
subnet {{vlan.cf.vlan_prefix}} {{vlan.cf.subnet_mask}}
{
       authoritative;
       allow unknown-clients;
       option domain-name-servers {{vlan.cf.domain_name_servers}};
       option routers {{vlan.cf.routers}};
       option domain-name "{{vlan.cf.domain_name}}";
       option time-offset {{vlan.cf.time_offset}};
       option broadcast-address {{vlan.cf.broadcast_address}};
       option subnet-mask {{vlan.cf.subnet_mask}};
       option slp-directory-agent true {{vlan.cf.slp_directory_agent}};
       option slp-service-scope true {{vlan.cf.slp_service_scope}};
       option netbios-name-servers {{vlan.cf.netbios_name_servers}};
       option netbios-dd-server {{vlan.cf.netbios_dd_server}};
       option netbios-node-type {{vlan.cf.netbios_node_type}};

       max-lease-time {{vlan.cf.max_lease_time}};
       default-lease-time {{vlan.cf.default_lease_time}};

       range {{vlan.cf.range}};
}
{% endfor %}

**Comment by jeremystretch** [comment]
Export templates are intended to work primarily with a single type of object. For more complex queries, you're likely better off writing a [custom report](https://docs.netbox.dev/en/stable/customization/reports/) to generate the exact data you need.

**Reply by Decafpancakes** [reply]
I can get it to kind of work by outputting the prefix from the vlan but then cant get it to reference the IPs. I think its because I'm not sure what the object would be. 

{%- for vlan in queryset %}

{%- for prefix in vlan.prefixes.all() %}

{{ vlan.id }},{{ vlan.name }},{{prefix.prefix}}

{%- endfor %}

{%- endfor %}

**Reply by candlerb** [reply]
Are you looking to iterate over the individual IPAddress records within the Prefix? Try the method [`get_child_ips()`](https://github.com/netbox-community/netbox/blob/v3.2.4/netbox/ipam/models/ip.py#L486-L494):

```
{%- for vlan in queryset %}
{%- for prefix in vlan.prefixes.all() %}
{%- for ipaddress in prefix.get_child_ips() %}
{{ vlan.id }},{{ vlan.name }},{{prefix.prefix}},{{ipaddress.address.ip}}
{%- endfor %}
{%- endfor %}
{%- endfor %}
```

**Reply by Decafpancakes** [reply]
Thank you so much! This is exactly what I needed!
